yash.c

```diff
@@ -59,6 +59,8 @@ static void execute_profile(const wchar_t *profile);
 static void execute_rcfile(const wchar_t *rcfile);
 static bool execute_file_in_home(const wchar_t *path)
     __attribute__((nonnull));
+static bool execute_file_in_xdg_config_home(const wchar_t *path)
+    __attribute__((nonnull));
```

Comment:
Can we merge these two functions into one?

```suggestion
static bool execute_file_in(const wchar_t *dir_var_name, const wchar_t *path)
    __attribute__((nonnull));
```

Assuming `dir_var_name` to be either `L VAR_XDG_CONFIG_HOME` or `L VAR_HOME`.